x-pack/plugins/triggers_actions_ui/public/application/components/json_editor_with_message_variables.tsx

```diff
@@ -44,17 +45,28 @@ export const JsonEditorWithMessageVariables: React.FunctionComponent<Props> = ({
   helpText,
   onBlur,
 }) => {
-  const [cursorPosition, setCursorPosition] = useState<any>(null);
+  const editorRef = useRef<monaco.editor.IStandaloneCodeEditor>();
+  const editorDisposables = useRef<monaco.IDisposable[]>([]);
 
   const { convertToJson, setXJson, xJson } = useXJsonMode(inputTargetValue ?? null);
 
   const onSelectMessageVariable = (variable: ActionVariable) => {
+    const editor = editorRef.current;
+    if (!editor) {
+      return;
+    }
```

Comment:
What would the UX be if the monaco reference is lost for some reason?
Will it just render an empty space?

Reply:
Yea good question, so this is mainly here to appease typescript's null checks (instead of doing reference checks all the time). 

In theory this would never be called if `editor` reference is unavailable since we maintain the reference as soon as the component mounts. There isn't any way to lose the reference right now. 

But yea if somehow we lost a reference, this function would no-op and the contents within the editor would remain unchanged

Reply:
Fair enough :)
In that case, I think we should throw, so that **if** an edgecase happens we'll at least show the user some feedback :)